lib/client/redirect.go

```diff
@@ -75,10 +75,14 @@ type Redirector struct {
 	context context.Context
 	// cancel broadcasts cancel
 	cancel context.CancelFunc
+	// issueSSOLoginConsoleRequest if not nil will be used as custom implementation of IssueSSOLoginConsoleRequest function.
+	issueSSOLoginConsoleRequest IssueSSOLoginConsoleRequest
 }
 
+type IssueSSOLoginConsoleRequest func(req SSOLoginConsoleReq) (*SSOLoginConsoleResponse, error)
```

Comment:
Nit: I think these types of things are usually named with "fn" or "func" suffix, I'd name it:
```suggestion
	// ssoLoginConsoleRequestFn allows to override SSO login request creation logic. Used for xxxxx.
	ssoLoginConsoleRequestFn SSOLoginConsoleRequestFunc
}

type SSOLoginConsoleRequestFunc func(req SSOLoginConsoleReq) (*SSOLoginConsoleResponse, error)
```

Reply:
Done: f0503dbbae9cbb592c3d347ae51b54c0b2f95235